packages/core/src/browser/menu/browser-main-menu-factory.ts

```diff
@@ -62,9 +71,17 @@ export class BrowserMainMenuFactory {
             this.fillMenuBar(menuBar);
         });
         menuBar.disposed.connect(() => listener.dispose());
+        this._menuBar = menuBar;
         return menuBar;
     }
 
+    protected update(): void {
```

Comment:
I think it should go rather to `BrowserMenuBarContribution` since it is responsible to manager it. The factory is responsible only to create a new instance.

Reply:
It would be good also to move `BrowserMenuBarContribution` to own file, i.e. it could stay in `browser-menu-plugin` to avoid breaking clients. It should be a primary API for clients to access menuBar, factory is implementation details.

Reply:
> it could stay in browser-menu-plugin to avoid breaking clients.

I want to do it before 1.0, and keep the module names in sync with the naming conventions. I do not mind if it breaks now, we have breaking changes in each release.

Reply:
It is minor, do as you feel right